### ISSUE TITLE
Ice lake generated near desert area

### ISSUE THREAD

**Issue by devolution79** [We-the-People-civ4col-mod/Mod#804]
develop 878f12eb3bdc0c93750f9930bac91dd42e1a42ad ( last commit as of 10.04.23 )

![image](https://user-images.githubusercontent.com/32330976/230850803-7df571a0-ce9b-4434-b4cf-59374b515733.png)

Note this was the first map I generated so not sure about the probability of this occuring. The default settings were used, size=huge

**Comment by raystuttgart** [comment]
That is caused by the fact that they connect to "Permafrost". :)
(This will trigger the Ice Lake)

The script actually geneartes a very strange map, when it generates "Desert" near "Permafrost".
So it is more of an issue of the MapScript and not really of the "Ice Lake" conversion logic.

"Desert" and "Permafrost" should really not be generated that close to each other.

-----

The "Ice Lake" conversion logic works perfectly fine in all "handcrafted" Maps.
But as it seems in MapScripts there are strange scenarios, where it does not.

**Comment by MrZorG33** [comment]
Yes, this is an incorrect operation of the map generator. there is still a lot of work on it.

**Comment by Nightinggale** [comment]
Technically speaking deserts don't have to be hot. The [Gobi Desert](https://en.wikipedia.org/wiki/Gobi_Desert) has an annual mean temperature of −2.5 °C.

Having said that, I did ponder about this issue when I touched upon the code recently. I did decide to stick to the same game logic and just fix the potential OOS issue, but I did wonder if just one plot of permafrost should be enough to freeze the entire lake regardless of size and other terrain.
